src/main/java/edu/hm/hafner/analysis/parser/SimulinkCheckParser.java

```diff
@@ -0,0 +1,116 @@
+            Document document = Jsoup.parse(file, "utf-8");
+            Elements systemElement = document.select("div.ReportContent");
+            String system = systemElement.first().id();
+            Report report = new Report();
+
+            //Select Warning items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.WarningCheck");
+            //Select Failed items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.FailedCheck");
+            //Select Not Run items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.NotRunCheck");
+            //Select Incomplete items from the whole html document
+            parseIssue(report, document, issueBuilder, system, "div.IncompleteCheck");
+
+            return report;
+        }
+        catch (IOException e) {
+            throw new ParsingException(e);
+        }
+    }
+
+    private void parseIssue(final Report report, final Document document, final IssueBuilder issueBuilder, final String system, final String check) {
+
+        Elements issueElements = document.select(check);
+        for (Element w : issueElements) {
+            Elements heading;
+            heading = w.select("span.CheckHeading");
+            String issueTxt;
+            issueTxt = heading.text();
+            setSeverity(check, issueBuilder);
+
+            String headingElement;
+            headingElement = heading.first().id();
+            String[] headingSplit;
+            headingSplit = headingElement.split("\\.");
+            int strLen;
+            strLen = headingSplit.length;
+            String module;
+
+            if (issueTxt.contains("SW0")) {
+                String[] fileName = issueTxt.split(":", 2);
+                module = headingSplit[strLen - 1];
+                issueBuilder.setModuleName(fileName[0] + "." + module);
+                issueBuilder.setDescription(fileName[1]);
+            }
+            else if (w.parent().id().contains("SW0")) {
+                String parentTxt = w.parent().id();
+                int i = parentTxt.indexOf("SW0");
+                parentTxt = parentTxt.substring(i);
+                module = headingSplit[strLen - 1];
+                issueBuilder.setModuleName(parentTxt + "." + module);
+                issueBuilder.setDescription(issueTxt);
+            }
+            else {
+                module = headingSplit[strLen - 1] + "." + headingSplit[strLen - 2];
+                issueBuilder.setModuleName(module);
+                issueBuilder.setDescription(issueTxt);
+                issueBuilder.setAdditionalProperties(w.parent().id());
+            }
+            issueBuilder.setFileName(system);
+            report.add(issueBuilder.build());
+        }
+    }
+
+    private void setSeverity(final String check, final IssueBuilder issueBuilder) {
+
+        if ("div.WarningCheck".equals(check)) {
```

Comment:
The String is already used above: use a constant

Reply:
Fixed